[WIP] Add CRUD API and Storage

teferi opened the pull request.

teferi committed:
Add raml API specification

teferi committed:
Add API skeleton and config management

boris-42 commented on runbook/mapping/es.py:
```diff
@@ -0,0 +1,69 @@
+# Copyright 2016: Mirantis Inc.
```
Let's move this module to: "runbook/storage.py" I like how it's done in Availability service (and actually we should do the same in health service)

teferi committed:
Add es storage and initialisation

This change includes initial mappings, config for backend es and code,
that initializes es indices if they're not present

teferi committed:
Implement runbook CRUD API

It is now possible to create, update and delete runbooks in
elasticsearch. Input validation is done with jsonschema.

teferi committed:
Add API tests

Also fix result conversion, from always returning the same runbook

teferi committed:
Alter Dockerfile to run the api and pre-create indices for es
